modules/ort/src/main/java/org/eclipse/sw360/antenna/ort/workflow/analyzers/OrtResultAnalyzer.java

```diff
@@ -111,209 +58,65 @@ public String getName() {
         while (scanResultsIterator.hasNext()) {
-            artifacts.add(mapArtifact(Optional.empty(), Optional.of(scanResultsIterator.next())));
+            OrtScannerResultResolver ortScannerResultResolver = new OrtScannerResultResolver(scanResultsIterator.next());
+            artifacts.add(ortScannerResultResolver.mapArtifact());
         }
         return artifacts;
     }
 
-    private Optional<JsonNode> getScanResultWithId(Optional<JsonNode> ortPackage, Optional<JsonNode> scanResult) {
-        Optional<JsonNode> pkgId = ortPackage.map(p -> p.get("id"));
+    private Optional<JsonNode> getScanResultWithId(JsonNode ortPackage, Optional<JsonNode> scanResult) {
+        JsonNode pkgId = ortPackage.get("id");
 
         return scanResult.flatMap(jsonNode -> StreamSupport.stream(jsonNode.spliterator(), false)
-                .filter(currentScanResult -> currentScanResult.get("id").equals(pkgId.get()))
+                .filter(currentScanResult -> currentScanResult.get("id").equals(pkgId))
                 .findFirst());
     }
 
-    private Artifact mapArtifact(Optional<JsonNode> ortPackage, Optional<JsonNode> scanResult) {
-        Artifact a;
-        if (ortPackage.isPresent()) {
-            a = mapArtifactFromAnalyzerResult(ortPackage.get());
-        } else if (scanResult.isPresent()) {
-            a = mapArtifactFromScanResult(scanResult.get());
-        } else {
-            return new Artifact("OrtResult");
-        }
+    private Artifact mapAnalyzerArtifact(JsonNode analyzerResult, Optional<JsonNode> scanResult) {
+        OrtAnalyzerResultResolver ortAnalyzerResultResolver = new OrtAnalyzerResultResolver(analyzerResult);
+        Artifact artifact = ortAnalyzerResultResolver.mapArtifact();
 
-        if (scanResult.isPresent()) {
-            LicensePair artifactLicenseCopyrightInfo = mapObservedLicense(scanResult.get());
-            a.addFact(new ObservedLicenseInformation(artifactLicenseCopyrightInfo.getLicenseInformation()))
-                    .addFact(artifactLicenseCopyrightInfo.getCopyrightStatement());
-        }
-
-        a.addFact(new ArtifactMatchingMetadata(MatchState.EXACT));
-
-        return a;
-    }
-
-    private Artifact mapArtifactFromAnalyzerResult(JsonNode ortPackage) {
-        Artifact artifact = new Artifact("OrtResult")
-                .addFact(mapFileName(ortPackage))
-                .addFact(new DeclaredLicenseInformation(mapDeclaredLicenses("declared_licenses", ortPackage)))
-                .addFact(new ArtifactSourceUrl(mapSourceUrl(ortPackage)));
-        mapCoordinates(ortPackage.get("id").textValue()).ifPresent(artifact::addFact);
-        mapHomepageUrl(ortPackage).ifPresent(homepageUrl -> artifact.addFact(new ArtifactHomepage(homepageUrl)));
+        scanResult.ifPresent(scnRslt -> {
+            OrtScannerResultResolver ortScannerResultResolver = new OrtScannerResultResolver(scnRslt);
+            Artifact artifact1 = ortScannerResultResolver.mapArtifact();
```

Comment:
maybe better without the introduction of a temporary variable
```suggestion
            ortScannerResultResolver.mapArtifact()
                .askForAll(ArtifactFact.class)
                .forEach(artifact::addFact);
```